src/v2g_exi_codec_gen/SchemaAnalyzer.py

```diff
@@ -546,9 +546,44 @@ def __get_particle_list(self, element: XsdElement, subst_list):
 
     @staticmethod
     def __add_choice_info_if_exists(element: XsdElement, elem_data: ElementData):
+        def sub_manipulate_element_data(model_group: XsdGroup):
+            elem_data.has_choice = True
+
+            choice = Choice()
+            for index, group in enumerate(model_group.iter_model()):
+                if hasattr(group, "model"):
```

Comment:
@chausGit

Reply:
Yes, you are right @a-w50, we could use alternatively isInstance().
We could even reduce this to isinstance(group, XsdGroup) instead of hasattr, since only the XsdGroup has the model attribute.